portfolio/src/main/java/com/google/sps/servlets/Background.java

```diff
@@ -0,0 +1,58 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//This file requests usernames and displays a welcome message.
+
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import com.google.gson.Gson;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/background")
+public class Background extends HttpServlet {
+
+  private String name = "";
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+   
+    response.setContentType("text/html;");
+    response.getWriter().println(name);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      String firstnme = retrievefstName(request, "firstname", "Anakin");
```

Comment:
These abbreviations feel unusual (I don't think this is common in javascript). I would recommend something like:

firstName

In general I think you want camel case for variables and you want it to be something you can search for (think if you were reviewing this in several months and you needed to search for this because you knew you used a pattern somewhere. Would you remember that you had called it ''firstnme"?